src/legacy/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -90,27 +85,22 @@ function getClauseForType(schema, namespace, type) {
 
 /**
  *  Get the "query" related keys for the search body
- *  @param  {EsMapping} mapping mappings from Ui
- * *@param  {SavedObjectsSchema} schema
- *  @param  {(string|Array<string>)} type
- *  @param  {String} search
- *  @param  {Array<string>} searchFields
- *  @param  {String} defaultSearchOperator
- *  @param  {Object} hasReference
- *  @return {Object}
  */
 export function getQueryParams(
-  mappings,
-  schema,
-  namespace,
-  type,
-  search,
-  searchFields,
-  defaultSearchOperator,
-  hasReference
+  mappings: IndexMapping,
+  schema: SavedObjectsSchema,
+  namespace?: string,
+  type?: string | string[],
+  search?: string,
+  searchFields?: string[],
+  defaultSearchOperator?: string,
+  hasReference?: {
+    type: string;
+    id: string;
+  }
 ) {
   const types = getTypes(mappings, type);
-  const bool = {
+  const bool: any = {
```

Comment:
Not sure if we should TSify bool queries in this PR?

Reply:
how deep we can define types? do we already have types describing  ES syntax?

Reply:
We don't current have types describing elasticsearch query syntax, I did a quick search through our code and `@types/elasticsearch` but they all use `any` or nothing.

@epixa thoughts on if we should or shouldn't create an interface for elasticsearch queries?

Reply:
We have no precedent for typing elasticsearch query DSL as far as I know. Unless you're planning to export this on a public interface, we can probably live without typing it.

Reply:
I will keep it as is for now then 👍